types/openebs.go

```diff
@@ -0,0 +1,234 @@
+// pools should be installed by default or not
+type SparsePools struct {
+	Enabled string `json:"enabled"`
+}
+
+// Provisioner stores the configuration of OpenEBS provisioner
+type Provisioner struct {
+	ComponentCommonConfig `json:",inline"`
+	Container             `json:",inline"`
+}
+
+// LocalProvisioner stores the configuration of OpenEBS local provisioner
+type LocalProvisioner struct {
+	ComponentCommonConfig `json:",inline"`
+	Container             `json:",inline"`
+}
+
+// AdmissionServer stores the configuration of openebs-admission-server
+type AdmissionServer struct {
+	ComponentCommonConfig `json:",inline"`
+	Container             `json:",inline"`
+}
+
+// SnapshotOperator stores the configuration of Snapshot Operator
+type SnapshotOperator struct {
+	ComponentCommonConfig `json:",inline"`
+	Controller            Container `json:"controller"`
+	Provisioner           Container `json:"provisioner"`
+}
+
+// NDM stores the configuration of Node disk Manager
+type NDM struct {
+	ComponentCommonConfig `json:",inline"`
+	Container             `json:",inline"`
+	Sparse                *Sparse     `json:"sparse"`
+	Filters               *NDMFilters `json:"filters"`
+	Probes                *NDMProbes  `json:"probes"`
+}
+
+// Sparse file configuration for NDM.
+type Sparse struct {
+	Path  string `json:"path"`
+	Size  string `json:"size"`
+	Count string `json:"count"`
+}
+
+// NDMFilters stores the configuration for filters being used by NDM
+// i.e., filters contain the config for excluding or including vendors,
+// paths, etc.
+type NDMFilters struct {
+	OSDisk *FilterConfigs `json:"OSDisk"`
+	Vendor *FilterConfigs `json:"vendor"`
+	Path   *FilterConfigs `json:"path"`
+}
+
+// FilterConfigs contains the config for NDM filters
+type FilterConfigs struct {
+	Enabled string  `json:"enabled"`
+	Include *string `json:"include"`
+	Exclude *string `json:"exclude"`
+}
+
+// NDMProbes can be used to configure NDM probes i.e., it can be used to
+// enable/disable various probes used by NDM such as seachest, smart,
+// capacity, etc.
+type NDMProbes struct {
+	Udev     *ProbeState `json:"udev"`
+	Smart    *ProbeState `json:"smart"`
+	Seachest *ProbeState `json:"seachest"`
+}
+
+// ProbeState denotes the current state of a NDM probe
+type ProbeState struct {
+	Enabled string `json:"enabled"`
+}
+
+// NDMOperator stores the configuration of NDM operator
+type NDMOperator struct {
+	ComponentCommonConfig `json:",inline"`
+	Container             `json:",inline"`
+}
+
+// Jiva stores the configuration of jiva.
+type Jiva struct {
+	ComponentCommonConfig `json:",inline"`
+	Container             `json:",inline"`
+}
+
+// Cstor stores the configuration of cstor.
+type Cstor struct {
+	Pool       Container `json:"pool"`
+	PoolMgmt   Container `json:"poolMgmt"`
+	Target     Container `json:"target"`
+	VolumeMgmt Container `json:"volumeMgmt"`
+}
+
+// Container stores the details of a container
+type Container struct {
```

Comment:
Future: IMO we should reuse Kubernetes Container. We can copy paste the entire struct if that helps us avoid lots of dependencies.